src/box/applier.cc

```diff
@@ -2840,7 +2859,8 @@ applier_on_state_f(struct trigger *trigger, void *event)
 	/* Wake up waiter */
 	fiber_cond_signal(&on_state->wakeup);
 
-	applier_pause(applier);
+	if (applier->state == on_state->desired_state)
+		applier_pause(applier);
```

Comment:
Maybe, better make `applier_pause` and `applier_resume` no-op in case the fiber is in the 'stopped' or 'off' state? It'd make the code look more robust.

Reply:
Discussed f2f and decided to drop this change as it not required for fixing the issue.